core/src/transport/upgrade.rs

```diff
@@ -113,6 +114,53 @@ where
             version,
         ))
     }
+
+    /// Upgrades the transport to perform authentication of the remote
+    ///
+    /// The supplied upgrade receives the I/O resource `C` and must
+    /// produce a pair `(PeerId, D)`, where `D` is a new I/O resource.
+    /// The upgrade must thus at a minimum identify the remote, which typically
+    /// involves the use of a cryptographic authentication protocol in the
+    /// context of establishing a secure channel.
+    ///
+    /// ## Transitions
+    ///
+    ///   * I/O upgrade: `C -> (PeerId, D)`.
+    ///   * Transport output: `C -> (PeerId, D)`
+    ///
+    /// ## Alternative state
+    ///
+    /// This function is an alternative code path to [`Builder::authenticate`]
+    /// given the supplied upgrade implements `InboundConnectionUpgrade`/`OutboundSecurityUpgrade`
```

Comment:
```suggestion
    /// given the supplied upgrade implements `InboundSecurityUpgrade`/`OutboundSecurityUpgrade`
```

Shouldn't it be this?

Also, can we make these rustdoc links please using `[]` to ensure they get checked at compile-time!

Reply:
Yes, I missed that :smile: